src/pages/experiments/[experimentId]/data-exploration/index.jsx

```diff
@@ -39,12 +40,22 @@ const ExplorationViewPage = () => {
   const router = useRouter();
   const { experimentId } = router.query;
   const layout = useSelector((state) => state.layout);
+  const embeddingState = useSelector((state) => state.embeddings)
+  const cellSetsState = useSelector((state) => state.cellSets)
   const { windows, panel } = layout;
   const [selectedTab, setSelectedTab] = useState(panel);
 
   useEffect(() => {
     setSelectedTab(panel);
-  }, [panel]);
+
+    // Toggle Louvain color when embedding and cellSets finish loading
+    if(embeddingState.umap !== undefined) {
+      if(!embeddingState.umap.loading && !cellSetsState.loading) {
+        dispatch(setCellInfoFocus(experimentId,'cellSets', 'louvain'))
+      }
+    }
+
+  }, [panel, embeddingState, cellSetsState]);
```

Comment:
I see what you are doing here and the reason why, but I don't think that it is a good approach - it makes the `ExplorationViewPage` tightly coupled with the cell sets and the embedding component. Deciding what keys to be checked and what not is internal to the cell sets tool component, the `ExplorationViewPage` shouldn't really know or care about this. My view is that it would be much better to dispatch the `setCellInfoFocus` action from `CellSetsTool` instead